docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+
+Step 2: `CommandParser#parse` would identify the user input as an Add Item Command by searching for the keyword `Add Item` within the user input. It would then remove `Add Item` from the user input causing the user input to be just `Toilet Roll / 15`. `CommandParser#parse` would then call the constructor of `AddItemCommand` and pass it `Toilet Roll / 15`.  
+
+Step 3:  An `AddItemCommand` object would be created. The constructor of `AddItemCommand` would call `AddItemCommand#extractItemPax` and `AddItemCommand#extractItemName` respectively to extract out `15` and `Toilet Roll`. The constructor of `AddItemCommand` would then call the constructor of `Item` which would create an `Item` object containing `Toilet Roll` and `15`. The `Item` object would then be returned to the constructor of `AddItemCommand`. The `Item` object would be saved to the `AddItemCommand` object using `AddItemCommand#setItem`. The `AddItemCommand` object would be passed back to `CommandParser#parse`.   
+
+Step 4: `CommandParser#parse` would then call `AddItemCommand#execute` which would call `AddItemCommand#getItem` to extract out the item currently saved within the `AddItemCommand` object.  
+
+Step 5: `AddItemCommand#execute` would then call `ItemList#addItemToList` and pass it the item object.This results in the `Item` object being added into the `ItemList`.
+
+Step 6: `AddItemCommand#execute` would then call `Ui#printAddItemAcknowledgementMessage` which would print out an acknowledgement message informing the user that the item has been succesfully added into the item list.
+
+### check room information by level
+
+The checking room information by level mechanism is facilitated `CheckRoomByLevelCommand`. It extends command. Additionally, it implements the following operations:
+* `CheckRoomByLevelCommand#isValidLevel(int level, RoomList roomList)` - Checks if the level number is valid
+* `CheckRoomByLevelCommand#printRoom(int level, RoomList roomlist)` - Prints all room information ar target level
+
+
+Given below is an example usage scenario of how `CheckRoomByLevelCommand` behaves at each step.
+
+Step 1: User launches the application for the first time. The status of all room will be vacant
+
+Step 2: User give a command `check level 2`. The `CommandParser` runs phase which will return a new `CheckRoomByLevelCommand`.
+and it will parse the parameter after replace command `check level` to ``.
+
+Step 3: The `CheckRoomByLevelCommand` will now contain `2`.
+
+Step 4: The `execute` method will call `isValidLevel(int level, RoomList roomList)` method to check if the level number is valid.
+
+Step 5: The `excute` method will then call `printRoom(int level, RoomList roomList)` method to print room information at target level.
+
+![sequence diagram](team/xunyi_checkroombylevelcommand_uml/checkRoomByLevel_Squence.png)
```

Comment:
Consider adding the activation bar in each class. Besides, some variable does not show its usage after assigning value to them (e.g., `isValidLevel`). Consider adding `opt` block for this variable.